### ISSUE TITLE
MTL-LoRA

### ISSUE THREAD

**Issue by FrLdy** [adapter-hub/adapters#793]
# 🌟 New adapter setup

## Model description

[Low-Rank Updates of pre-trained Weights for Multi-Task Learning](https://aclanthology.org/2023.findings-acl.476/) 

Multi-Task Learning used with pre-trained models has been quite popular in the field of Natural Language Processing in recent years. This framework remains still challenging due to the complexity of the tasks and the challenges associated with fine-tuning large pre-trained models. In this paper, we propose a new approach for Multi-task learning which is based on stacking the weights of Neural Networks as a tensor. We show that low-rank updates in the canonical polyadic tensor decomposition of this tensor of weights lead to a simple, yet efficient algorithm, which without loss of performance allows to reduce considerably the model parameters. We investigate the interactions between tasks inside the model as well as the inclusion of sparsity to find the best tensor rank and to increase the compression rate. Our strategy is consistent with recent efforts that attempt to use constraints to fine-tune some model components. More precisely, we achieve equivalent performance as the state-of-the-art on the General Language Understanding Evaluation benchmark by training only 0.3 of the parameters per task while not modifying the baseline weights.

## Open source status

* [ ] the model implementation is available: No
* [ ] the model weights are available: No
* [ ] who are the authors: @audibert-alexandre-fra
